config/config/examples/derive.rs

```diff
@@ -0,0 +1,88 @@
+//! Run with: `cargo run --example derive`
+//! Look at the generated code with: `cargo expand --example derive`
+
+use config::{sources, Config, ConfigBuilder, ConfigError};
+
+type TypeAlias = bool;
+
+#[derive(config::Config, Debug, PartialEq, serde::Deserialize, Default)]
+#[serde(default)]
+struct AppConfig {
+    enabled: TypeAlias,
+    logging: LoggingConfig,
+    #[config(cli(skip), env(skip))]
+    count: Vec<Vec<u8>>,
+}
+
+#[derive(config::Config, Debug, PartialEq, serde::Deserialize)]
+#[serde(default)]
+struct LoggingConfig {
+    level: String,
+    json: bool,
+}
+
+impl Default for LoggingConfig {
+    fn default() -> Self {
+        Self {
+            level: "INFO".to_string(),
+            json: false,
+        }
+    }
+}
+
+fn main() {
+    match parse() {
+        Ok(config) => println!("{:#?}", config),
+        Err(err) => println!("{:#}", err),
+    }
+}
+
+fn parse() -> Result<AppConfig, ConfigError> {
+    dbg!(AppConfig::graph());
+
+    let mut builder = ConfigBuilder::new();
+    builder.add_source(sources::CliSource::new()?);
+    builder.add_source(sources::EnvSource::with_prefix("TEST")?);
+    builder.add_source(sources::FileSource::json(
+        br#"
+    {
+        "enabled": "on",
+        "count": [[2], [122]],
+        "logging": {
+            "level": "DEBUG",
+            "json": true
+        }
+    }
+    "#
+        .as_slice(),
+    )?);
+
+    let config: AppConfig = builder.build()?;
+
+    Ok(config)
+}
+
+#[derive(Debug, Default, serde::Deserialize, config::Config)]
+#[serde(default)]
+struct AppConfig2 {
```

Comment:
What is the purpose of this struct?

Reply:
We don't need it anymore. It's still there from earlier tests I think.